Implement live routes

Abyss777 opened the pull request:
- Live routes are drawn always, the button just toggles layer visibility
- Tails length can be configured via `web.liveRouteLength` server or user attribute.
- Added wrapper `getAttributePreference` to use it in future

![image](https://cloud.githubusercontent.com/assets/5688080/19755817/b70a16d6-9c31-11e6-8c99-658a2a9fc512.png)

Abyss777 commented on web/app/view/MapController.js:
```diff
@@ -166,6 +182,43 @@ Ext.define('Traccar.view.MapController', {
+
+                if (deviceId in this.liveRoutes) {
+                    lastLiveCoordinates = this.liveRoutes[deviceId][this.liveRoutes[deviceId].length - 1].getGeometry().getCoordinates();
+                    if (lastLiveCoordinates[0] === geometry.getCoordinates()[0] &&
```
This `if` prevent tails cuts every 30 seconds in debug mode because of websocket re-connection. Also it is useful if device send positions (with same coordinates) while parking.
Only thing is `jscs` complains about `continue` statement.

What should I do?
- Remove workaround
- Refactor code to avoid `continue`
- Configure `jscs` to not complain about `continue`

tananaev replied:
I think we should get rid of "continue". Also, it would be nice to split this huge method into separate smaller methods for more readability.

tananaev commented on web/app/Style.js:
```diff
@@ -77,5 +77,7 @@ Ext.define('Traccar.Style', {
-    reportTagfieldWidth: 375
+    reportTagfieldWidth: 375,
+
+    headerButtonsMargin: '0 5'
```
What is this for?

Abyss777 replied:
Five is left margin, without this it look like:
![image](https://cloud.githubusercontent.com/assets/5688080/19794823/043f14ac-9cef-11e6-9424-c1d3fc3c4e59.png)

tananaev commented on web/app/view/MapController.js:
```diff
@@ -166,6 +182,43 @@ Ext.define('Traccar.view.MapController', {
+                    });
+                    liveLine.setStyle(this.getRouteStyle(position.get('deviceId')));
+                    this.liveRoutes[deviceId].push(liveLine);
+                    this.getView().getLiveRouteSource().addFeature(liveLine);
```
Do we add line for every position item? Can't we extend existing line instead?

Abyss777 commented:
Split `updateLatest` function
Used one `lineString` 
I was thinking, If we do not react on marker click it can confuse user, I just removed markers. Looks good:

![image](https://cloud.githubusercontent.com/assets/5688080/19797011/bcda5e10-9d01-11e6-8bdf-f1278e21f6a0.png)

Also `continue` become `return` after splitting.

tananaev commented:
Looks really good, thanks.
